[CI] MixedClusterClientYamlTestSuiteIT test {p0=cat.templates/10_basic/Filtered templates} failing

bwc test for 8.14.0 failing with:
```
legacy template [v1_template] has index patterns [t*, t1*] matching patterns from existing composable templates [traces-apm.rum@template,traces-apm.sampled@template,traces-apm@template] with patterns (traces-apm.rum@template => [traces-apm.rum-*],traces-apm.sampled@template => [traces-apm.sampled-*],traces-apm@template => [traces-apm-*]), use composable templates (/_index_template) instead
```

**Build scan:**
https://gradle-enterprise.elastic.co/s/l2nmmkwrk7jcm/tests/:qa:mixed-cluster:v8.14.0%23mixedClusterTest/org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT/test%20%7Bp0=cat.templates%2F10_basic%2FFiltered%20templates%7D


**Reproduction line:**
```
./gradlew ':qa:mixed-cluster:v8.14.0#mixedClusterTest' -Dtests.class="org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=cat.templates/10_basic/Filtered templates}" -Dtests.seed=91DBDF4EA89D20E0 -Dtests.bwc=true -Dtests.locale=sr -Dtests.timezone=SystemV/AST4 -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
[Failure dashboard for `org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT#test {p0=cat.templates/10_basic/Filtered templates}`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Bp0%3Dcat.templates/10_basic/Filtered%20templates%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: Failure at [cat.templates/10_basic:89]: expected [2xx] status code but api [indices.put_template] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"legacy template [test] has index patterns [t*] matching patterns from existing composable templates [traces-apm.rum@template,traces-apm.sampled@template,traces-apm@template] with patterns (traces-apm.rum@template => [traces-apm.rum-*],traces-apm.sampled@template => [traces-apm.sampled-*],traces-apm@template => [traces-apm-*]), use composable templates (/_index_template) instead","stack_trace":"org.elasticsearch.ElasticsearchException$1: legacy template [test] has index patterns [t*] matching patterns from existing composable templates [traces-apm.rum@template,traces-apm.sampled@template,traces-apm@template] with patterns (traces-apm.rum@template => [traces-apm.rum-*],traces-apm.sampled@template => [traces-apm.sampled-*],traces-apm@template => [traces-apm-*]), use composable templates (/_index_template) instead\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:704)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:632)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.build(RestResponse.java:184)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:143)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:114)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.rest.action.RestActionListener.onFailure(RestActionListener.java:53)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:401)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.tasks.TaskManager$1.onFailure(TaskManager.java:214)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:317)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:401)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:317)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:401)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$2(TransportMasterNodeAction.java:234)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.acceptException(ActionListenerImplementations.java:186)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onFailure(ActionListenerImplementations.java:191)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.admin.indices.template.put.TransportPutIndexTemplateAction$1.onFailure(TransportPutIndexTemplateAction.java:107)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$TemplateClusterStateUpdateTask.onFailure(MetadataIndexTemplateService.java:172)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$ExecutionResult.notifyFailure(MasterService.java:986)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$ExecutionResult.onClusterStateUnchanged(MasterService.java:949)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:246)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.lambda$run$2(MasterService.java:1656)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListener.run(ActionListener.java:433)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.run(MasterService.java:1653)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$5.lambda$doRun$0(MasterService.java:1248)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListener.run(ActionListener.java:433)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$5.doRun(MasterService.java:1227)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1570)\nCaused by: java.lang.IllegalArgumentException: legacy template [test] has index patterns [t*] matching patterns from existing composable templates [traces-apm.rum@template,traces-apm.sampled@template,traces-apm@template] with patterns (traces-apm.rum@template => [traces-apm.rum-*],traces-apm.sampled@template => [traces-apm.sampled-*],traces-apm@template => [traces-apm-*]), use composable templates (/_index_template) instead\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.innerPutTemplate(MetadataIndexTemplateService.java:1164)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$7.execute(MetadataIndexTemplateService.java:1112)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:148)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:145)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.SimpleBatchedExecutor.execute(SimpleBatchedExecutor.java:70)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:1050)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:1015)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:233)\n\t... 11 more\n"}],"type":"illegal_argument_exception","reason":"legacy template [test] has index patterns [t*] matching patterns from existing composable templates [traces-apm.rum@template,traces-apm.sampled@template,traces-apm@template] with patterns (traces-apm.rum@template => [traces-apm.rum-*],traces-apm.sampled@template => [traces-apm.sampled-*],traces-apm@template => [traces-apm-*]), use composable templates (/_index_template) instead","stack_trace":"java.lang.IllegalArgumentException: legacy template [test] has index patterns [t*] matching patterns from existing composable templates [traces-apm.rum@template,traces-apm.sampled@template,traces-apm@template] with patterns (traces-apm.rum@template => [traces-apm.rum-*],traces-apm.sampled@template => [traces-apm.sampled-*],traces-apm@template => [traces-apm-*]), use composable templates (/_index_template) instead\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.innerPutTemplate(MetadataIndexTemplateService.java:1164)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$7.execute(MetadataIndexTemplateService.java:1112)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:148)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:145)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.SimpleBatchedExecutor.execute(SimpleBatchedExecutor.java:70)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:1050)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:1015)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:233)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.lambda$run$2(MasterService.java:1656)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListener.run(ActionListener.java:433)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.run(MasterService.java:1653)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$5.lambda$doRun$0(MasterService.java:1248)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListener.run(ActionListener.java:433)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$5.doRun(MasterService.java:1227)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1570)\n"},"status":400}]

  at __randomizedtesting.SeedInfo.seed([91DBDF4EA89D20E0:198FE09406614D18]:0)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:610)
  at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:553)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1570)

  Caused by: java.lang.AssertionError: expected [2xx] status code but api [indices.put_template] returned [400 Bad Request] [{"error":{"root_cause":[{"type":"illegal_argument_exception","reason":"legacy template [test] has index patterns [t*] matching patterns from existing composable templates [traces-apm.rum@template,traces-apm.sampled@template,traces-apm@template] with patterns (traces-apm.rum@template => [traces-apm.rum-*],traces-apm.sampled@template => [traces-apm.sampled-*],traces-apm@template => [traces-apm-*]), use composable templates (/_index_template) instead","stack_trace":"org.elasticsearch.ElasticsearchException$1: legacy template [test] has index patterns [t*] matching patterns from existing composable templates [traces-apm.rum@template,traces-apm.sampled@template,traces-apm@template] with patterns (traces-apm.rum@template => [traces-apm.rum-*],traces-apm.sampled@template => [traces-apm.sampled-*],traces-apm@template => [traces-apm-*]), use composable templates (/_index_template) instead\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.ElasticsearchException.guessRootCauses(ElasticsearchException.java:704)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.ElasticsearchException.generateFailureXContent(ElasticsearchException.java:632)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.build(RestResponse.java:184)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:143)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.rest.RestResponse.<init>(RestResponse.java:114)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.rest.action.RestActionListener.onFailure(RestActionListener.java:53)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:401)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.tasks.TaskManager$1.onFailure(TaskManager.java:214)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:317)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:401)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.DelegatingActionListener.onFailure(DelegatingActionListener.java:31)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$RunBeforeActionListener.onFailure(ActionListenerImplementations.java:317)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeOnFailure(ActionListenerImplementations.java:73)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListener$3.onFailure(ActionListener.java:401)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$2(TransportMasterNodeAction.java:234)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.acceptException(ActionListenerImplementations.java:186)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations.safeAcceptException(ActionListenerImplementations.java:62)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListenerImplementations$DelegatingResponseActionListener.onFailure(ActionListenerImplementations.java:191)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.admin.indices.template.put.TransportPutIndexTemplateAction$1.onFailure(TransportPutIndexTemplateAction.java:107)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$TemplateClusterStateUpdateTask.onFailure(MetadataIndexTemplateService.java:172)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$ExecutionResult.notifyFailure(MasterService.java:986)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$ExecutionResult.onClusterStateUnchanged(MasterService.java:949)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:246)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.lambda$run$2(MasterService.java:1656)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListener.run(ActionListener.java:433)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.run(MasterService.java:1653)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$5.lambda$doRun$0(MasterService.java:1248)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListener.run(ActionListener.java:433)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$5.doRun(MasterService.java:1227)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1570)\nCaused by: java.lang.IllegalArgumentException: legacy template [test] has index patterns [t*] matching patterns from existing composable templates [traces-apm.rum@template,traces-apm.sampled@template,traces-apm@template] with patterns (traces-apm.rum@template => [traces-apm.rum-*],traces-apm.sampled@template => [traces-apm.sampled-*],traces-apm@template => [traces-apm-*]), use composable templates (/_index_template) instead\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.innerPutTemplate(MetadataIndexTemplateService.java:1164)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$7.execute(MetadataIndexTemplateService.java:1112)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:148)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:145)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.SimpleBatchedExecutor.execute(SimpleBatchedExecutor.java:70)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:1050)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:1015)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:233)\n\t... 11 more\n"}],"type":"illegal_argument_exception","reason":"legacy template [test] has index patterns [t*] matching patterns from existing composable templates [traces-apm.rum@template,traces-apm.sampled@template,traces-apm@template] with patterns (traces-apm.rum@template => [traces-apm.rum-*],traces-apm.sampled@template => [traces-apm.sampled-*],traces-apm@template => [traces-apm-*]), use composable templates (/_index_template) instead","stack_trace":"java.lang.IllegalArgumentException: legacy template [test] has index patterns [t*] matching patterns from existing composable templates [traces-apm.rum@template,traces-apm.sampled@template,traces-apm@template] with patterns (traces-apm.rum@template => [traces-apm.rum-*],traces-apm.sampled@template => [traces-apm.sampled-*],traces-apm@template => [traces-apm-*]), use composable templates (/_index_template) instead\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService.innerPutTemplate(MetadataIndexTemplateService.java:1164)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$7.execute(MetadataIndexTemplateService.java:1112)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:148)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.metadata.MetadataIndexTemplateService$1.executeTask(MetadataIndexTemplateService.java:145)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.SimpleBatchedExecutor.execute(SimpleBatchedExecutor.java:70)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:1050)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:1015)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:233)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.lambda$run$2(MasterService.java:1656)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListener.run(ActionListener.java:433)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.run(MasterService.java:1653)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$5.lambda$doRun$0(MasterService.java:1248)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.action.ActionListener.run(ActionListener.java:433)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.cluster.service.MasterService$5.doRun(MasterService.java:1227)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)\n\tat org.elasticsearch.server@8.15.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\tat java.base/java.lang.Thread.run(Thread.java:1570)\n"},"status":400}]

    at org.junit.Assert.fail(Assert.java:89)
    at org.elasticsearch.test.rest.yaml.section.DoSection.checkResponseException(DoSection.java:540)
    at org.elasticsearch.test.rest.yaml.section.DoSection.execute(DoSection.java:396)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.executeSection(ESClientYamlSuiteTestCase.java:590)
    at org.elasticsearch.test.rest.yaml.ESClientYamlSuiteTestCase.test(ESClientYamlSuiteTestCase.java:553)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke(Method.java:580)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.junit.rules.RunRules.evaluate(RunRules.java:20)
    at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at org.junit.rules.RunRules.evaluate(RunRules.java:20)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
    at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
    at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
    at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
    at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
    at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
    at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
    at org.junit.rules.RunRules.evaluate(RunRules.java:20)
    at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
    at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
    at java.lang.Thread.run(Thread.java:1570)

```

--- a/docs/reference/indices/put-index-template.asciidoc
+++ b/docs/reference/indices/put-index-template.asciidoc
@@ -12,7 +12,7 @@ that can be applied automatically to new indices.
 --------------------------------------------------
 PUT /_index_template/template_1
 {
-  "index_patterns" : ["te*"],
+  "index_patterns" : ["template*"],
   "priority" : 1,
   "template": {
     "settings" : {
@@ -186,7 +186,7 @@ You can include <<aliases,index aliases>> in an index template.
 --------------------------------------------------
 PUT _index_template/template_1
 {
-  "index_patterns" : ["te*"],
+  "index_patterns" : ["template*"],
   "template": {
     "settings" : {
         "number_of_shards" : 1
@@ -218,7 +218,7 @@ the template with the highest priority is used. For example:
 --------------------------------------------------
 PUT /_index_template/template_1
 {
-  "index_patterns" : ["t*"],
+  "index_patterns" : ["temp*"],
   "priority" : 0,
   "template": {
     "settings" : {
@@ -233,7 +233,7 @@ PUT /_index_template/template_1
 
 PUT /_index_template/template_2
 {
-  "index_patterns" : ["te*"],
+  "index_patterns" : ["template*"],
   "priority" : 1,
   "template": {
     "settings" : {
@@ -246,7 +246,7 @@ PUT /_index_template/template_2
 }
 --------------------------------------------------
 
-For indices that start with `te*`, `_source` will enabled, and the index will have two primary
+For indices that start with `template*`, `_source` will enabled, and the index will have two primary
 shards and one replica, because only `template_2` will be applied.
 
 NOTE: Multiple templates with overlapping index patterns at the same priority are not allowed, and

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/cat.templates/10_basic.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/cat.templates/10_basic.yml
@@ -92,7 +92,7 @@
           body:
             order: 0
             version: 1
-            index_patterns: t*
+            index_patterns: test*
             settings:
               number_of_shards: 1
               number_of_replicas: 0
@@ -103,7 +103,7 @@
           body:
             order: 2
             version: 1
-            index_patterns: tea*
+            index_patterns: nomatch*
             settings:
               number_of_shards: 1
               number_of_replicas: 0
@@ -116,7 +116,7 @@
         $body: |
                 /^
                     test    \s+
-                    \[t\*\] \s+
+                    \[test\*\] \s+
                     0       \s+
                     1       \s*
                     \n
@@ -134,7 +134,7 @@
           body:
             order: 0
             version: 1
-            index_patterns: t*
+            index_patterns: test*
             settings:
               number_of_shards: 1
               number_of_replicas: 0
@@ -154,7 +154,7 @@
                     composed_of
                     \n
                     test            \s+
-                    \[t\*\]         \s+
+                    \[test\*\]         \s+
                     0               \s+
                     1               \s*
                     \n
@@ -172,7 +172,7 @@
           body:
             order: 0
             version: 1
-            index_patterns: t*
+            index_patterns: test*
             settings:
               number_of_shards: 1
               number_of_replicas: 0
@@ -190,7 +190,7 @@
                     index_patterns
                     \n
                     test        \s+
-                    \[t\*\]
+                    \[test*\*\]
                     \n
                 $/
 
@@ -206,7 +206,7 @@
           name: test
           body:
             order: 0
-            index_patterns: t*
+            index_patterns: test*
             settings:
               number_of_shards: 1
               number_of_replicas: 0
@@ -217,7 +217,7 @@
           body:
             order: 0
             version: 1
-            index_patterns: te*
+            index_patterns: test-*
             settings:
               number_of_shards: 1
               number_of_replicas: 0
@@ -230,8 +230,8 @@
     - match:
         $body: |
               /^
-                  test   \s+ \[t\*\]  \s+   \n \n
-                  test_1 \s+ \[te\*\] \s+ 1 \n \n
+                  test   \s+ \[test\*\]  \s+   \n \n
+                  test_1 \s+ \[test-\*\] \s+ 1 \n \n
               $/
 
     - do:
@@ -242,8 +242,8 @@
     - match:
         $body: |
               /^
-                  test_1 \s+ \[te\*\]   \s+ 1\n \n
-                  test   \s+ \[t\*\]    \s+  \n \n
+                  test_1 \s+ \[test-\*\]   \s+ 1\n \n
+                  test   \s+ \[test\*\]    \s+  \n \n
 
               $/
 
@@ -260,7 +260,7 @@
           body:
             order: 0
             version: 1
-            index_patterns: [t*, te*]
+            index_patterns: [test*, test-*]
             settings:
               number_of_shards: 1
               number_of_replicas: 0
@@ -278,7 +278,7 @@
                  index_patterns
                  \n
                  test_1       \s+
-                 \[t\*,\ te\*\]
+                 \[test\*,\ test-\*\]
                  \n
                  \n
               $/

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.simulate_index_template/10_basic.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.simulate_index_template/10_basic.yml
@@ -9,7 +9,7 @@
       indices.put_index_template:
         name: test
         body:
-          index_patterns: te*
+          index_patterns: test*
           template:
             settings:
               number_of_shards:   1
@@ -39,7 +39,7 @@
       indices.put_index_template:
         name: existing_test
         body:
-          index_patterns: te*
+          index_patterns: test*
           priority: 10
           template:
             settings:
@@ -66,7 +66,7 @@
       indices.simulate_index_template:
         name: test
         body:
-          index_patterns: te*
+          index_patterns: test*
           priority: 15
           template:
             settings:
@@ -79,7 +79,7 @@
   - match: {template.settings.index.number_of_replicas: "2"}
   - match: {template.mappings.properties.ct_field.type: "keyword"}
   - match: {overlapping.0.name: existing_test}
-  - match: {overlapping.0.index_patterns: ["te*"]}
+  - match: {overlapping.0.index_patterns: ["test*"]}
   - length: {template.aliases: 1}
   - is_true: template.aliases.test_alias
 
@@ -92,11 +92,11 @@
 
   - do:
       allowed_warnings:
-        - "index template [test] has index patterns [te*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [test] will take precedence during new index creation"
+        - "index template [test] has index patterns [test*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [test] will take precedence during new index creation"
       indices.put_index_template:
         name: test
         body:
-          index_patterns: te*
+          index_patterns: test*
           priority: 10
           template:
             settings:
@@ -124,18 +124,18 @@
       indices.put_template:
         name: v1_template
         body:
-          index_patterns: [t*, t1*]
+          index_patterns: [test*, t1*]
           settings:
             number_of_shards:   5
 
   - do:
       allowed_warnings:
-        - "index template [v2_template] has index patterns [te*] matching patterns from existing older templates [v1_template] with patterns
-        (v1_template => [t*, t1*]); this template [v2_template] will take precedence during new index creation"
+        - "index template [v2_template] has index patterns [test*] matching patterns from existing older templates [v1_template] with patterns
+        (v1_template => [test*, t1*]); this template [v2_template] will take precedence during new index creation"
       indices.put_index_template:
         name: v2_template
         body:
-          index_patterns: te*
+          index_patterns: test*
           priority: 10
           template:
             settings:
@@ -148,12 +148,12 @@
 
   - do:
       allowed_warnings:
-        - "index template [winning_v2_template] has index patterns [te*] matching patterns from existing older templates [v1_template] with patterns
-        (v1_template => [t*, t1*]); this template [winning_v2_template] will take precedence during new index creation"
+        - "index template [winning_v2_template] has index patterns [test*] matching patterns from existing older templates [v1_template] with patterns
+        (v1_template => [test*, t1*]); this template [winning_v2_template] will take precedence during new index creation"
       indices.put_index_template:
         name: winning_v2_template
         body:
-          index_patterns: te*
+          index_patterns: test*
           priority: 20
           template:
             settings:
@@ -172,9 +172,9 @@
   - match: {template.settings.index.number_of_replicas: "0"}
   - match: {template.mappings.properties.field.type: "keyword"}
   - match: {overlapping.0.name: v1_template}
-  - match: {overlapping.0.index_patterns: ["t*", "t1*"]}
+  - match: {overlapping.0.index_patterns: ["test*", "t1*"]}
   - match: {overlapping.1.name: v2_template}
-  - match: {overlapping.1.index_patterns: ["te*"]}
+  - match: {overlapping.1.index_patterns: ["test*"]}
 
 ---
 "Simulate an index for and index or alias that already exists":
@@ -187,7 +187,7 @@
       indices.put_index_template:
         name: test
         body:
-          index_patterns: [te*]
+          index_patterns: [test*]
           template:
             settings:
               number_of_shards:   1
@@ -235,7 +235,7 @@
       indices.put_index_template:
         name: test
         body:
-          index_patterns: te*
+          index_patterns: test*
           template:
             lifecycle:
               data_retention: "7d"

--- a/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.simulate_template/10_basic.yml
+++ b/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.simulate_template/10_basic.yml
@@ -39,7 +39,7 @@
       indices.put_index_template:
         name: existing_test
         body:
-          index_patterns: te*
+          index_patterns: test*
           priority: 10
           template:
             settings:
@@ -65,7 +65,7 @@
   - do:
       indices.simulate_template:
         body:
-          index_patterns: te*
+          index_patterns: test*
           priority: 15
           template:
             settings:
@@ -78,7 +78,7 @@
   - match: {template.settings.index.number_of_replicas: "2"}
   - match: {template.mappings.properties.ct_field.type: "keyword"}
   - match: {overlapping.0.name: existing_test}
-  - match: {overlapping.0.index_patterns: ["te*"]}
+  - match: {overlapping.0.index_patterns: ["test*"]}
   - length: {template.aliases: 1}
   - is_true: template.aliases.test_alias
 
@@ -93,18 +93,18 @@
       indices.put_template:
         name: v1_template
         body:
-          index_patterns: [t*, t1*]
+          index_patterns: [test*, t1*]
           settings:
             number_of_shards:   5
 
   - do:
       allowed_warnings:
-        - "index template [v2_template] has index patterns [te*] matching patterns from existing older templates [v1_template] with patterns
-        (v1_template => [t*, t1*]); this template [v2_template] will take precedence during new index creation"
+        - "index template [v2_template] has index patterns [test*] matching patterns from existing older templates [v1_template] with patterns
+        (v1_template => [test*, t1*]); this template [v2_template] will take precedence during new index creation"
       indices.put_index_template:
         name: v2_template
         body:
-          index_patterns: te*
+          index_patterns: test*
           priority: 10
           template:
             settings:
@@ -117,12 +117,12 @@
 
   - do:
       allowed_warnings:
-        - "index template [winning_v2_template] has index patterns [te*] matching patterns from existing older templates [v1_template] with patterns
-        (v1_template => [t*, t1*]); this template [winning_v2_template] will take precedence during new index creation"
+        - "index template [winning_v2_template] has index patterns [test*] matching patterns from existing older templates [v1_template] with patterns
+        (v1_template => [test*, t1*]); this template [winning_v2_template] will take precedence during new index creation"
       indices.put_index_template:
         name: winning_v2_template
         body:
-          index_patterns: te*
+          index_patterns: test*
           priority: 20
           template:
             settings:
@@ -141,9 +141,9 @@
   - match: {template.settings.index.number_of_replicas: "0"}
   - match: {template.mappings.properties.field.type: "keyword"}
   - match: {overlapping.0.name: v1_template}
-  - match: {overlapping.0.index_patterns: ["t*", "t1*"]}
+  - match: {overlapping.0.index_patterns: ["test*", "t1*"]}
   - match: {overlapping.1.name: v2_template}
-  - match: {overlapping.1.index_patterns: ["te*"]}
+  - match: {overlapping.1.index_patterns: ["test*"]}
 
 ---
 "Simulate replacing a template with a newer version":
@@ -156,7 +156,7 @@
       indices.put_index_template:
         name: v2_template
         body:
-          index_patterns: te*
+          index_patterns: test*
           priority: 10
           template:
             settings:
@@ -183,7 +183,7 @@
       indices.simulate_template:
         name: v2_template
         body:
-          index_patterns: te*
+          index_patterns: test*
           priority: 10
           template:
             settings:

